### PR TITLE
fix: find waiting for cy.intercept when it has been overwritten by the user

### LINKED ISSUE #9580: Alias for intercept is not working if overwriting cy.intercept command
Cypress v6.1.0

```js
describe('intercept', () => {
  Cypress.Commands.overwrite('intercept', (intercept, ...args) => {
    return cy.log('intercept!').then(() => {
      return intercept(...args)
    })
  })

  it('is registered too late', () => {
    cy.visit('/')
    cy.intercept('/todos').as('todos')
    cy.wait('@todos')
  })
})
```

The alias `todos` is not recognized, but the routes table is showing it

Screen Shot 2020-12-08 at 9 35 57 PM

Note: even putting `cy.wait` into `cy.then` does not solve it - something is really off about this alias.

### PR DESCRIPTION
**Pull request by bahmutov**
- Closes #9580

### User facing changelog

If the user overwrites `cy.intercept`, it still works with `cy.wait(alias)`

This used to show an error on `cy.wait`, and now it works

```js
Cypress.Commands.overwrite('intercept', function (originalIntercept, ...args) {
  return cy.log('intercept!').then(() => {
    return originalIntercept(...args)
  })
})

cy.intercept('/foo', 'my value').as('netAlias')
cy.wait('@netAlias')
```

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub?

### CODE REVIEW

**Comment by bahmutov** [comment]
Notes: I built this PR on top of https://github.com/cypress-io/cypress/pull/9589 but probably will have clean merge

Also this PR will need reconcile with https://github.com/cypress-io/cypress/pull/9594

File: packages/driver/src/cy/commands/waiting.js [hunk 1]
```diff
@@ -162,11 +162,28 @@ module.exports = (Commands, Cypress, cy, state) => {
...
+        const commandName = command.get('name')
+
+        return commandsThatCreateNetworkIntercepts.includes(commandName)
+      }
+
+      const findNetworkAlias = (alias) => {
+        const routes = cy.state('routes') || {}
+
+        return _.find(_.values(routes), { alias })
+      }
```
Comment (flotwig) [inline packages/driver/src/cy/commands/waiting.js:176]: this doesn't cover dynamic aliases with `req.alias` i don't think

but luckily you can use `getAliasedRequests`: https://github.com/cypress-io/cypress/blob/c6f03e390f2b3c55f9c3b9a08611c6aabad2d607/packages/driver/src/cy/net-stubbing/aliasing.ts#L14-L28

**Review by flotwig — CHANGES_REQUESTED** [review]
nice

File: packages/driver/src/cy/commands/waiting.js (hunk 1 above)
Comment (bahmutov) [reply packages/driver/src/cy/commands/waiting.js:170]: Seems to work as is, I have added a test to confirm it

**Review by flotwig — DISMISSED** [review]
Does not fix #3890, but as #3890 is getting deprecated anyways, I'll allow it, LGTM

**Pull request merged**